### ISSUE TITLE
Combining HuSpaCy models with pipe objects having n_process > 1 results in error.

### ISSUE THREAD

**Issue by ViktorVarga29** [huspacy/huspacy#66]
**Describe the bug**
Trying to use a spaCy pipe object to take advantage of multiprocessing for turning docs into tokenized, processed versions I get an error when using a combination of bigger than 1 values for the n_process argument of pipe and one of the huspacy models.
As a test I have tried both using the said Hungarian models with n_process=1 and using a default English model of spaCy with higher n_process values which both succeed. 

Traceback:
```

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[16], line 1
----> 1 texts = docs_to_texts(t_docs)

Cell In[15], line 8, in docs_to_texts(docs)
      4 nlp = spacy.load("hu_core_news_lg", exclude=['tok2vec', 'senter', 'tagger', 'morphologizer', 'parser'])
      6 texts = []
----> 8 for doc in nlp.pipe(docs, n_process=4):
      9     with doc.retokenize() as retokenizer:
     10         for ent in doc.ents:

File /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/spacy/language.py:1618, in Language.pipe(self, texts, as_tuples, batch_size, disable, component_cfg, n_process)
   1616     for pipe in pipes:
   1617         docs = pipe(docs)
-> 1618 for doc in docs:
   1619     yield doc

File /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/spacy/language.py:1701, in Language._multiprocessing_pipe(self, texts, pipes, n_process, batch_size)
   1699 elif byte_error is not None:
   1700     error = srsly.msgpack_loads(byte_error)
-> 1701     self.default_error_handler(
   1702         None, None, None, ValueError(Errors.E871.format(error=error))
   1703     )
   1704 if i % batch_size == 0:
   1705     # tell `sender` that one batch was consumed.
   1706     sender.step()

File /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/spacy/util.py:1704, in raise_error(proc_name, proc, docs, e)
   1703 def raise_error(proc_name, proc, docs, e):
-> 1704     raise e

ValueError: [E871] Error encountered in nlp.pipe with multiprocessing:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/spacy/language.py", line 2332, in _apply_pipes
    byte_docs = [(doc.to_bytes(), doc._context, None) for doc in docs]
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/spacy/language.py", line 2332, in <listcomp>
    byte_docs = [(doc.to_bytes(), doc._context, None) for doc in docs]
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/spacy/util.py", line 1685, in _pipe
    yield from proc.pipe(docs, **kwargs)
  File "spacy/pipeline/transition_parser.pyx", line 245, in pipe
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/spacy/util.py", line 1632, in minibatch
    batch = list(itertools.islice(items, int(batch_size)))
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/spacy/util.py", line 1685, in _pipe
    yield from proc.pipe(docs, **kwargs)
  File "spacy/pipeline/trainable_pipe.pyx", line 79, in pipe
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/spacy/util.py", line 1704, in raise_error
    raise e
  File "spacy/pipeline/trainable_pipe.pyx", line 75, in spacy.pipeline.trainable_pipe.TrainablePipe.pipe
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/hu_core_news_lg/edit_tree_lemmatizer.py", line 192, in predict
    scores = self.model.predict(docs)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/thinc/model.py", line 334, in predict
    return self._func(self, X, is_train=False)[0]
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/thinc/layers/chain.py", line 54, in forward
    Y, inc_layer_grad = layer(X, is_train=is_train)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/thinc/model.py", line 310, in __call__
    return self._func(self, X, is_train=is_train)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/thinc/layers/chain.py", line 54, in forward
    Y, inc_layer_grad = layer(X, is_train=is_train)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/thinc/model.py", line 310, in __call__
    return self._func(self, X, is_train=is_train)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/thinc/layers/chain.py", line 54, in forward
    Y, inc_layer_grad = layer(X, is_train=is_train)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/thinc/model.py", line 310, in __call__
    return self._func(self, X, is_train=is_train)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/thinc/layers/concatenate.py", line 57, in forward
    Ys, callbacks = zip(*[layer(X, is_train=is_train) for layer in model.layers])
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/thinc/layers/concatenate.py", line 57, in <listcomp>
    Ys, callbacks = zip(*[layer(X, is_train=is_train) for layer in model.layers])
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/thinc/model.py", line 310, in __call__
    return self._func(self, X, is_train=is_train)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/spacy/ml/staticvectors.py", line 56, in forward
    V = vocab.vectors.get_batch(keys)
  File "spacy/vectors.pyx", line 485, in spacy.vectors.Vectors.get_batch
  File "spacy/strings.pyx", line 176, in spacy.strings.StringStore.as_string
  File "spacy/strings.pyx", line 160, in spacy.strings.StringStore.__getitem__
KeyError: "[E018] Can't retrieve string for hash '2911968107040651156'. This usually refers to an issue with the `Vocab` or `StringStore`."
```

**To Reproduce**
- I am using Jupyter Notebook for the project
- Python 3.10
- spaCy 3.7.2
- huspacy 0.11.0
- hu_core_news_lg 3.7.0 / hu_core_news_md 3.7.0
This is the involved part of the code:

```python
def docs_to_texts(docs:list[str]) -> list[list[str]]:
    docs = [remove_affixes(doc, affixes) for doc in docs]
    
    nlp = spacy.load("hu_core_news_lg", exclude=['tok2vec', 'senter', 'tagger', 'morphologizer', 'parser'])
    
    texts = []
    
    for doc in nlp.pipe(docs, n_process=4):
        with doc.retokenize() as retokenizer:
            for ent in doc.ents:
                retokenizer.merge(doc[ent.start:ent.end])
        texts.append([token.lemma_ for token in doc)
            
    return texts

texts = docs_to_texts(t_docs)
```

**Expected behavior**
I aimed to use the pipe object to speed up the otherwise very speed consuming iteration over thousands of documents.

**Comment by oroszgy** [comment]
Hi @ViktorVarga29, thanks for the detailed report. I am happy to help, but was not able to reproduce your error. (Tried in [this Colab Notebook](https://gist.github.com/oroszgy/fb409992dfd2c02579f39dccc14cab0a)). Could you please share a fully reproducible and environment so that I can work on this issue?

**Comment by ViktorVarga29** [comment]
Hi @oroszgy, thank you for the rapid response. I went into further testing using the simplified minimal version of code, used in your Colab Notebook in different environments. I have used different combinations of package installations, python versions testing both as standalone scripts and in cells of Jupyter notebooks.

The results were very inconsistent. In some cases succeeding as a script and failing in a notebook using the same environment, or succeeding in case of using the _hu_core_news_md_ model, but failing for the _hu_core_news_lg_ model.

As the Colab Notebook you've provided has also failed for me, which has its own environment, I assumed it is a problem specific to the underlying OS/hardware and how it handles multiprocessing. (MacBook Pro M1, macOS 14.0)

The solution that resulted in consistent success both under python 3.10.12 and 3.12 using latest versions of spacy, huspacy and the language models was to manually set the multiprocessing start method to 'fork' instead of the default 'spawn' before initiating the pipe processing.
```
if __name__=='__main__':
    multiprocessing.set_start_method('fork')
    texts = docs_to_texts(docs)
```
I'm not exactly sure what's the reason and now I think this problem is not specific to these models and huspacy and it happened by chance that the English model I tried at first has worked.
Thank you for your assistance anyway!
